.github/workflows/container-build.yaml

```diff
@@ -41,38 +42,23 @@ jobs:
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
 
-      - uses: actions/setup-node@v3
-        with:
-          node-version: ${{ env.NODE_VERSION }}
-
       - name: Prepare
         id: prep
         run: .github/workflows/prepare.sh
         env:
           KIND: ${{ matrix.kind }}
           APP: ${{ matrix.app }}
 
-      - name: Pre-build artifacts using Containerfile
+      - name: Build frontend
         uses: redhat-actions/buildah-build@v2
-        if: ${{ steps.prep.outputs.pre_build == 'containerfile' }}
         with:
-          image: builder
+          image: localhost/openshift-knative/frontend
           tags: ${{ github.sha }}
-          containerfiles: ${{ steps.prep.outputs.containerfile_prebuild }}
-          platforms: linux/amd64
-          context: ${{ matrix.app }}
+          containerfiles: frontend/deployment/Containerfile
+          platforms: ${{ steps.prep.outputs.platforms }}
```

Comment:
This build is simplified, but it's not optimal.

Now we are building both frontend and backend for every platform we support. This takes a long time, especially on emulated platforms.

In theory, we could just build it once (on `linux/amd64`), and then just create a simple multi-arch image from the build artifacts. Both JVM, and TS/JS support the "build once, run everywhere" feature.

Unfortunately, it is not possible with this approach of using `FROM ... as frontend/builder`, and `COPY --from=builder ...` because the container engine errors saying it doesn't have an image for given architecture.

Maybe we can achieve that by utilizing volumes, for passing down the artifacts. In that way, we can build frontend, and backend apps once and then just create a multi-arch container image.

Reply:
From my point of view, this could be merged as it is. It works, and corrected in follow up PR.